core/session.ts

```diff
@@ -392,6 +392,16 @@ export class Session {
       )
     );
 
+    this.checkNoDotsInTargetFields(
```

Comment:
why do we need to do this at the end of compilation? IMO it should be happening well before then - at action declaration time.

Reply:
Where are you suggesting to put it?

Would it be both:
* Action functions in the session https://github.com/dataform-co/dataform/blob/b5368e04a047c6f2c958b6dd14d9bd5f872e4e0f/core/session.ts#L236C11-L236C21
* Action constructors https://github.com/dataform-co/dataform/blob/b5368e04a047c6f2c958b6dd14d9bd5f872e4e0f/core/actions/table.ts#L291

Reply:
Everything passes through the action constructors, right? So we should only do it there.